infra/modules/service/secrets.tf

```diff
@@ -1,14 +1,14 @@
-data "aws_ssm_parameter" "secret" {
-  for_each = { for secret in var.secrets : secret.name => secret }
-  name     = each.value.ssm_param_name
-}
-
 locals {
   secrets = [
     for secret in var.secrets :
     {
       name      = secret.name,
-      valueFrom = data.aws_ssm_parameter.secret[secret.name].arn
+      valueFrom = secret.ssm_param_name
     }
   ]
+
+  secret_arn_patterns = [
+    for secret in var.secrets :
+    "arn:aws:ssm:*:*:parameter/${replace(secret.ssm_param_name, "/^//", "")}"
```

Comment:
I don't work with regex very often and it took me a moment to understand what is happening here. Would it be cleaner/clearer to use `trimprefix(secret.ssm_param_name, "/")` here?

Reply:
haha, when you have a solution that's obviously simpler and better, say it with confidence! i won't be offended :)

Reply:
k updated